src/_pytest/cacheprovider.py

```diff
@@ -33,22 +33,26 @@
 @attr.s
 class Cache(object):
     _cachedir = attr.ib(repr=False)
-    _warn = attr.ib(repr=False)
 
     @classmethod
     def for_config(cls, config):
         cachedir = cls.cache_dir_from_config(config)
         if config.getoption("cacheclear") and cachedir.exists():
             shutil.rmtree(str(cachedir))
             cachedir.mkdir()
-        return cls(cachedir, config.warn)
+        return cls(cachedir)
 
     @staticmethod
     def cache_dir_from_config(config):
         return paths.resolve_from_str(config.getini("cache_dir"), config.rootdir)
 
     def warn(self, fmt, **args):
-        self._warn(code="I9", message=fmt.format(**args) if args else fmt)
+        import warnings
```

Comment:
lets please make the imports toplevel and introduce a `PytestCacheWarning` which expresses the warnings of this subsystem

Reply:
@nicoddemus i beleive as a followup we should inline this method and adapt stacklevels, so people see the warnings located in their code

Reply:
> lets please make the imports toplevel and introduce a PytestCacheWarning which expresses the warnings of this subsystem

This is worth discussing a little: should we in general create a new warning class for each plugin, or just using `PytestWarning` is enough? 

My take is that just having a few warning classes is enough, I don't see much reason to have a ton of warning classes unless we have a good use case for that.

I ask because there's only 2 calls to `Config.warn`:

```python
self.warn("could not create cache path {path}", path=path)
...
self.warn("cache could not write path {path}", path=path)
```

Is `Config.warn` something that we meant to be public so others use it? It feels like it was made public without much thought behind it, mostly seems like an accident to me actually.

Reply:
@nicoddemus this self.warn can be completely replaced by a warnings.warn

i beleive we can remove it as internal api, if there turns out to be a user, we can bring it back